config/config.go

```diff
@@ -79,15 +81,74 @@ type Vehicle struct {
 	Host string `yaml:"host"`
 }
 
-// ReadConfig reads the config file from the given path
-func ReadConfig(configFiles embed.FS, configFileName string) (*Config, error) {
+// ReadConfig reads the configuration file from the embedded file system (configFiles),
+// fetches remote configuration from the provided URL (configURL), and merges them.
+func ReadConfig(configFiles embed.FS, configURL, configFileName string) (*Config, error) {
 	// read config file from embed.FS
 	data, err := fs.ReadFile(configFiles, configFileName)
 	if err != nil {
 		return nil, fmt.Errorf("failed to read config file: %w", err)
 	}
 
-	filePathOnDisk := "/opt/autopi/config.yaml"
+	configPathOnDisk := "/opt/autopi/config.yaml"
+	err = os.WriteFile(configPathOnDisk, data, 0644)
+	if err != nil {
+		return nil, fmt.Errorf("failed to write config file: %w", err)
+	}
+
+	config, err := ReadConfigFromPath(configPathOnDisk)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to read config file: %w", err)
+	}
+
+	// Get secrets from remote config
+	remoteConfigPathOnDisk := "/opt/autopi/remote-config.json"
+	remoteConfig, err := GetRemoteConfig(configURL, remoteConfigPathOnDisk)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to read secrets file: %w", err)
+	}
+
+	config.Services.Auth.ClientID = remoteConfig.Services.Auth.ClientID
+	config.Services.Auth.ClientSecret = remoteConfig.Services.Auth.ClientSecret
+	config.Services.Ca.CaFingerprint = remoteConfig.Services.Ca.CaFingerprint
+
+	return config, nil
+}
+
+// GetRemoteConfig sends a GET request to fetch the configuration and saves it to filePathOnDisk
+func GetRemoteConfig(configURL string, filePathOnDisk string) (*Config, error) {
+
+	// Check if the file already exists
+	_, err := os.Stat(filePathOnDisk)
+	if !os.IsNotExist(err) && err == nil {
+		return ReadConfigFromPath(filePathOnDisk)
+	}
+
+	if configURL == "" {
+		return nil, fmt.Errorf("configURL is empty")
+	}
+
+	// Send GET request
+	resp, err := http.Get(configURL)
```

Comment:
how should retries be handled? I think we had some shared code that had retries built in for http requests?

Reply:
yep, good point, just updated